app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,51 @@
+require_relative "../lib/oidc_client.rb"
+
+class SessionsController < ApplicationController
+  def create
+    if logged_in?
+      redirect_to redirect_path
+      return
+    end
+
+    redirect_to oidc.auth_uri[:uri]
+  end
+
+  def callback
+    unless params[:code]
+      redirect_to redirect_path
+      return
+    end
+
+    state = params.require(:state)
+
+    user_info = oidc.callback(
+      params.require(:code),
+      state,
+    )
+
+    session[:sub] = user_info.sub
+    session[:email] = user_info.email
+
+    redirect_to redirect_path
+  end
+
+  def delete
+    session.delete(:sub)
+    session.delete(:email)
+    redirect_to redirect_path
+  end
+
+private
+
+  def redirect_path
+    transition_checker_questions_path
+  end
+
+  def oidc
+    @oidc ||= OidcClient.new(
+      ENV["ACCOUNT_MANAGER_URL"],
+      ENV["OIDC_CLIENT_ID"],
+      ENV["OIDC_CLIENT_SECRET"],
```

Comment:
I'm wondering if there's anything we can do to bring these together with [these Variables](https://github.com/alphagov/govuk-docker/pull/417/files#diff-81716fac3ebcda04e6914c20d3333c82R38-R42):

... would the client_id always be the same between the JWT and the OIDC? If so could we merge them and just use `GOVUK_ACCOUNT_OAUTH_CLIENT_ID` perhaps?

The client secret will almost certainly be unique as we don't want to confuse the JWT permissions for these... :thinking: 

What's actually in there, is it a key?
I'm leaning towards OTT descriptive names for these, ones that flag this is for GOV.UK accounts and if it's a key say if it's public or not, or perhaps just secret if it's something else?

Those are all things I've scratched my head over a bit when digging through the env...

perhaps `GOVUK_ACCOUNT_OAUTH_CLIENT_SECRET`?

Reply:
You're absolutely right.  I've just realised `GOVUK_ACCOUNT_OAUTH_CLIENT_ID` and `OIDC_CLIENT_ID` are the same.  I'll update to use the more descriptive name that already exists, so `OIDC_CLIENT_ID` can be removed from https://github.com/alphagov/govuk-docker/pull/417.

Reply:
I'll also rename `OIDC_CLIENT_SECRET` to `GOVUK_ACCOUNT_OIDC_CLIENT_SECRET` as that is much clearer and consistent